src/services/IConfiguration.hh

```diff
@@ -20,8 +20,11 @@ enum class Feature
     PreferDecimal,
 };
 
-class IConfiguration {
+class IConfiguration
+{
 public:
+    virtual ~IConfiguration() {}
```

Comment:
this should be marked with `noexcept `and a `default`. Like this...

`virtual ~IConfiguration() noexcept = default;`

There's other stuff you should do too but don't worry about it, I was thinking of putting in a PR with proper/explicit constructors later.

Reply:
will do. this is mostly out of habit - you'll probably have to remind me again